packages/aws-cdk/test/integ/cli/bootstrapping.integtest.ts

```diff
@@ -26,7 +29,47 @@ test('can bootstrap without execution', async () => {
 
     expect(resp.Stacks?.[0].StackStatus).toEqual('REVIEW_IN_PROGRESS');
   } finally {
-    await deleteStacks(bootstrapStackName);
+    await deleteBootstrapStack(bootstrapStackName);
+  }
+});
+
+test('upgrade legacy bootstrap stack to new bootstrap stack while in use', async () => {
+  const bootstrapStackName = fullStackName('updating-bootstrap-stack');
+
+  const legacyBootstrapBucketName = `aws-cdk-bootstrap-integ-test-legacy-bckt-${randomString()}`;
+  const newBootstrapBucketName = `aws-cdk-bootstrap-integ-test-v2-bckt-${randomString()}`;
+
+  // Legacy bootstrap
+  await cdk(['bootstrap',
+    '--toolkit-stack-name', bootstrapStackName,
+    '--bootstrap-bucket-name', legacyBootstrapBucketName]);
+  try {
+    // Deploy stack that uses file assets
+    await cdkDeploy('lambda', {
+      options: ['--toolkit-stack-name', bootstrapStackName],
+    });
+
+    // Upgrade bootstrap stack to "new" style
+    await cdk(['bootstrap',
+      '--toolkit-stack-name', bootstrapStackName,
+      '--bootstrap-bucket-name', newBootstrapBucketName], {
+      modEnv: {
+        CDK_NEW_BOOTSTRAP: '1',
+      },
+    });
+
+    // (Force) deploy stack again
+    await cdkDeploy('lambda', {
+      options: [
+        '--toolkit-stack-name', bootstrapStackName,
+        '--force',
+      ],
+    });
+  } finally {
```

Comment:
Why use `finally`, instead of `afterEach` / `afterAll`, like before?

Reply:
Because we know the stack name here. I guess we could move to an `after` too...

Reply:
Oh it's because of what we need to destroy at the end of the test depends on the test, it's different every time.